### PR TITLE
feat: convert event_sub to http poll

### PR DESCRIPTION
**Pull request by sammy1991106**
## Todos

- [ ] Unit tests
- [ ] Manual tests
- [ ] Documentation
- [ ] Connect epics/issues
- [ ] Tag type of change

### CODE REVIEW

File: src/event_sub.rs [hunk 1]
```diff
@@ -95,49 +103,37 @@ impl<T: TmClient + Sync> EventSubClient<T> {
...
-
-                let mut sub = self
-                    .client
-                    .subscribe(EventType::NewBlock.into())
-                    .change_context(SubscriptionFailed)
-                    .await?;
+                let mut curr_block_height = match self.start_from {
+                    Some(start_from) => start_from,
+                    None => self.get_latest_block_height().await?,
+                };
```
Comment (cgorenflo) [inline src/event_sub.rs:109]:
```suggestion
                let mut curr_block_height = self.start_from.unwrap_or(self.get_latest_block_height().await?};
```

File: src/event_sub.rs [hunk 2]
```diff
@@ -95,49 +103,37 @@ impl<T: TmClient + Sync> EventSubClient<T> {
...
+    async fn get_latest_block_height(&self) -> Result<block::Height, EventSubError> {
+        Ok(self
+            .client
+            .latest_block()
+            .change_context(RPCFailed)
+            .await?
+            .block
+            .header()
+            .height)
+    }
```
Comment (cgorenflo) [inline src/event_sub.rs:135]:
```suggestion
    async fn get_latest_block_height(&self) -> Result<block::Height, EventSubError> {
        let response = self.client.latest_block().change_context(RPCFailed).await?;
        Ok( response.block.header().height)
    }
```

File: src/event_sub.rs [hunk 3]
```diff
@@ -200,170 +196,174 @@ pub enum EventSubError {
...
-    use tokio::test;
-
-    use crate::event_sub::{Event, EventSubClient, EventSubError};
-    use crate::tm_client;
+    use std::convert::TryInto;
+    use std::time::Duration;
+    use tendermint::block;
 
-    #[test]
+    #[tokio::test]
```
Comment (cgorenflo) [inline src/event_sub.rs:209]: why did you remove the import?

File: src/tm_client.rs [hunk 4]
```diff
@@ -1,48 +1,25 @@
...
 pub type BlockResultsResponse = tendermint_rpc::endpoint::block_results::Response;
 pub type BlockResponse = tendermint_rpc::endpoint::block::Response;
-pub type TxResponse = tendermint_rpc::endpoint::broadcast::tx_sync::Response;
 pub type Error = tendermint_rpc::Error;
-pub type Query = tendermint_rpc::query::Query;
-pub type Event = tendermint_rpc::event::Event;
-pub type EventData = tendermint_rpc::event::EventData;
-pub type EventType = tendermint_rpc::query::EventType;
 
 #[async_trait]
```
Comment (cgorenflo) [inline src/tm_client.rs:10]:
```suggestion
#[automock]
#[async_trait]
```

File: src/event_sub.rs [hunk 5]
```diff
@@ -200,170 +196,174 @@ pub enum EventSubError {
...
-        impl tm_client::TmClient for WebsocketClient{
-            type Sub = MockSubscription;
-
-            async fn subscribe(&self, query:tm_client::Query) -> Result<<Self as tm_client::TmClient>::Sub, tm_client::Error>;
+        impl tm_client::TmClient for TmClient{
             async fn latest_block(&self) -> Result<tm_client::BlockResponse, tm_client::Error>;
-            async fn block_results(&self, block_height: Height) -> Result<tm_client::BlockResultsResponse, tm_client::Error>;
-            fn close(self) -> Result<(), tm_client::Error>;
+            async fn block_results(&self, height: block::Height) -> Result<tm_client::BlockResultsResponse, tm_client::Error>;
         }
```
Comment (cgorenflo) [inline src/event_sub.rs:367]: without the associated type out of our control, the client can get automocked (see tm_client.rs), no need to define it manually anymore

File: src/event_sub.rs (hunk 1 above)
Comment (sammy1991106) [reply src/event_sub.rs:109]: I kinda didn't want it to make an extra RPC call when the `start_from` is given, but I don't have a very strong opinion on this.

File: src/event_sub.rs (hunk 3 above)
Comment (sammy1991106) [reply src/event_sub.rs:209]: oh I had another test that didn't necessarily have to be `tokio::test` but a regular sync one. I'll get it back.

File: src/event_sub.rs (hunk 5 above)
Comment (sammy1991106) [reply src/event_sub.rs:367]: oh nice!

**Review by cgorenflo — APPROVED** [review]
